DMCompiler/DM/Expressions/Unary.cs

```diff
@@ -123,4 +123,32 @@ public PostDecrement(Location location, DMExpression expr)
             proc.Decrement(reference);
         }
     }
+
+    // &x
+    internal sealed class PointerRef(Location location, DMExpression expr) : AssignmentUnaryOp(location, expr) {
```

Comment:
This and `PointerDeref` should inherit from `UnaryOp` instead since they don't do any assigning.